lib/internal/async_context_frame.js

```diff
@@ -41,6 +34,37 @@ class AsyncContextFrame extends Map {
     const frame = this.current();
     frame?.disable(store);
   }
+}
+
+function checkEnabled() {
+  const enabled = require('internal/options')
+    .getOptionValue('--experimental-async-context-frame');
+
+  // If enabled, swap to active prototype so we don't need to check status
+  // on every interaction with the async context frame.
+  if (enabled) {
+    // eslint-disable-next-line no-use-before-define
+    ObjectSetPrototypeOf(AsyncContextFrame, ActiveAsyncContextFrame);
```

Comment:
Is it possible to implement two separate variants and export the right one depending on the command line flag instead of using the prototype magic here?

Reply:
No. There's a timing problem that prevents loading the options interface that early.